src/main/java/edu/cornell/kfs/tax/dataaccess/impl/TransactionRowPRNCBuilder.java

```diff
@@ -0,0 +1,542 @@
+    // ==============================================================================================
+
+    /**
+     * Default implementation for building 1042S transaction detail rows from PRNC data.
+     */
+    static class For1042S extends TransactionRowPRNCBuilder<Transaction1042SSummary> {
+        private int numExcludedAssignedIncomeCodes;
+        private int numExcludedAssignedIncomeCodeSubTypes;
+        private int numDeterminedFederalTaxWithheldIncomeCodes;
+        private int numUndeterminedFederalTaxWithheldIncomeCodes;
+        private int numDeterminedStateIncomeTaxWithheldIncomeCodes;
+        private int numUndeterminedStateIncomeTaxWithheldIncomeCodes;
+        
+        For1042S() {
+            super();
+        }
+        
+        @Override
+        TaxTableField getExtraField(TransactionDetailRow detailRow) {
+            return detailRow.form1042SBox;
+        }
+        
+        @Override
+        String getTaxTypeSpecificConditionForSelect(Transaction1042SSummary summary) {
+            // No extra conditions needed to filter out DV rows for 1042S processing.
+            return KRADConstants.EMPTY_STRING;
+        }
+        
+        @Override
+        void doTaxSpecificRowSetup(ResultSet rs, PreparedStatement insertStatement, String financialObjectCode,
+                Transaction1042SSummary summary) throws SQLException {
+            TransactionDetailRow detailRow = summary.transactionDetailRow;
+            
+            // Setup prepared statement args accordingly. The 1042S box is set to "?" for bulk retrieval convenience.
+            insertStatement.setString(detailRow.incomeCode.index - detailRow.insertOffset, null);
+            insertStatement.setString(detailRow.incomeCodeSubType.index - detailRow.insertOffset, null);
+            insertStatement.setString(detailRow.form1099Box.index - detailRow.insertOffset, null);
+            insertStatement.setString(detailRow.form1099OverriddenBox.index - detailRow.insertOffset, null);
+            insertStatement.setString(detailRow.form1042SBox.index - detailRow.insertOffset, CUTaxConstants.NEEDS_UPDATING_BOX_KEY);
+            insertStatement.setString(detailRow.form1042SOverriddenBox.index - detailRow.insertOffset, null);
+        }
+        
+        @Override
+        void doTaxSpecificSecondPassRowSetup(ResultSet rs, Transaction1042SSummary summary) throws SQLException {
+            String financialObjectCode = rs.getString(summary.transactionDetailRow.finObjectCode.index);
+            String incomeClassCode;
+            String incomeCode;
+            String incomeCodeSubType;
+            
+            // Prepare the income and subtype codes.
+            incomeClassCode = rs.getString(summary.transactionDetailRow.incomeClassCode.index);
+            incomeCode = summary.incomeClassCodeToIrsIncomeCodeMap.get(incomeClassCode);
+            if (StringUtils.isBlank(incomeCode)) {
```

Comment:
Are you getting the expected 1042S row processing results for PRNC-derived rows? It looks like incomeClassCode is set to null when building the rows, yet this processing is potentially setting up undetermined-income-code rows unless they represent tax withholdings. You should verify with Cathy and Lori if the 1042S outcomes are what they expected; if they're not, then I would suggest using processing similar to that from TransactionRowPdpBuilder.For1042S for deriving the income class code from the object code or other related field.